In [54]:
# Libs

import pandas
import numpy as np
import matplotlib.pyplot as plt
import os


In [55]:
data_source = "../data/intact/"
data_filename = "bbs6-bbs10-bbs12-2024-11-08-20-14.txt"

data = pandas.read_csv(data_source + data_filename, sep="\t", header=0)

data.head()

ID(s) interactor A ID(s) interactor B  \
0   uniprotkb:Q8TAM1   uniprotkb:Q6ZW61   
1   uniprotkb:Q6ZW61   uniprotkb:Q8TAM1   
2   uniprotkb:Q8TAM1   uniprotkb:Q6ZW61   
3   uniprotkb:Q6ZW61   uniprotkb:Q8TAM1   
4   uniprotkb:Q8TAM1   uniprotkb:Q6ZW61   

                             Alt. ID(s) interactor A  \
0  uniprotkb:Q96CW2|uniprotkb:Q9H5D2|ensembl:ENSP...   
1  uniprotkb:Q8NAB8|uniprotkb:D3DNX5|uniprotkb:Q7...   
2  uniprotkb:Q96CW2|uniprotkb:Q9H5D2|ensembl:ENSP...   
3  uniprotkb:Q8NAB8|uniprotkb:D3DNX5|uniprotkb:Q7...   
4  uniprotkb:Q96CW2|uniprotkb:Q9H5D2|ensembl:ENSP...   

                             Alt. ID(s) interactor B  \
0  uniprotkb:Q8NAB8|uniprotkb:D3DNX5|uniprotkb:Q7...   
1  uniprotkb:Q96CW2|uniprotkb:Q9H5D2|ensembl:ENSP...   
2  uniprotkb:Q8NAB8|uniprotkb:D3DNX5|uniprotkb:Q7...   
3  uniprotkb:Q96CW2|uniprotkb:Q9H5D2|ensembl:ENSP...   
4  uniprotkb:Q8NAB8|uniprotkb:D3DNX5|uniprotkb:Q7...   

                              Alias(es) interactor A  \
0  psi-mi:bbs10_human(display_short)|psi-mi:Barde...   
1  psi-mi:bbs12_human(display_short)|psi-mi:Barde...   
2  psi-mi:bbs10_human(display_short)|psi-mi:Barde...   
3  psi-mi:bbs12_human(display_short)|psi-mi:Barde...   
4  psi-mi:bbs10_human(display_short)|psi-mi:Barde...   

                              Alias(es) interactor B  \
0  psi-mi:bbs12_human(display_short)|psi-mi:Barde...   
1  psi-mi:bbs10_human(display_short)|psi-mi:Barde...   
2  psi-mi:bbs12_human(display_short)|psi-mi:Barde...   
3  psi-mi:bbs10_human(display_short)|psi-mi:Barde...   
4  psi-mi:bbs12_human(display_short)|psi-mi:Barde...   

                    Interaction detection method(s) Publication 1st author(s)  \
0  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)       Seo S. et al.(2010)   
1  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)       Seo S. et al.(2010)   
2  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)       Seo S. et al.(2010)   
3  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)       Seo S. et al.(2010)   
4               psi-mi:"MI:0071"(molecular sieving)       Seo S. et al.(2010)   

                           Publication Identifier(s)  \
0  intact:EBI-15826730|doi:10.1073/pnas.091026810...   
1  intact:EBI-15826730|doi:10.1073/pnas.091026810...   
2  intact:EBI-15826730|doi:10.1073/pnas.091026810...   
3  intact:EBI-15826730|doi:10.1073/pnas.091026810...   
4  intact:EBI-15826730|doi:10.1073/pnas.091026810...   

                           Taxid interactor A  \
0  taxid:9606(human)|taxid:9606(Homo sapiens)   
1  taxid:9606(human)|taxid:9606(Homo sapiens)   
2  taxid:9606(human)|taxid:9606(Homo sapiens)   
3  taxid:9606(human)|taxid:9606(Homo sapiens)   
4  taxid:9606(human)|taxid:9606(Homo sapiens)   

                           Taxid interactor B  \
0  taxid:9606(human)|taxid:9606(Homo sapiens)   
1  taxid:9606(human)|taxid:9606(Homo sapiens)   
2  taxid:9606(human)|taxid:9606(Homo sapiens)   
3  taxid:9606(human)|taxid:9606(Homo sapiens)   
4  taxid:9606(human)|taxid:9606(Homo sapiens)   

                      Interaction type(s)  \
0  psi-mi:"MI:0915"(physical association)   
1           psi-mi:"MI:0914"(association)   
2  psi-mi:"MI:0915"(physical association)   
3           psi-mi:"MI:0914"(association)   
4  psi-mi:"MI:0915"(physical association)   

                                  Source database(s)  \
0  psi-mi:"MI:0465"(Database of Interacting Prote...   
1  psi-mi:"MI:0465"(Database of Interacting Prote...   
2  psi-mi:"MI:0465"(Database of Interacting Prote...   
3  psi-mi:"MI:0465"(Database of Interacting Prote...   
4  psi-mi:"MI:0465"(Database of Interacting Prote...   

                           Interaction identifier(s)  Confidence value(s)  
0  intact:EBI-15826767|dip:DIP-191289E|imex:IM-21...  intact-miscore:0.58  
1  intact:EBI-15826832|dip:DIP-191290E|imex:IM-21...  intact-miscore:0.58  
2  intact:EBI-15827036|dip:DIP-191289E|imex:IM-21...  intact-miscore:0.58  
3  intact:EBI-15827166|dip:DIP-191304E|imex:IM-21...  intact-miscore:

### <b>Columns removed from data</b>

We assume that both interactors B and A are from Taxon id 9606 (Homo sapiens) as this filter was applied when the search of data was performed, therefore we will exclude these columns "Taxid interactor A", "Taxid interactor B".

We also know the search was performed with the filter of interactor (A or B) as either of BBS6, BBS10 or BBS12, therefore we can remove the following columns: "Alt ID(s) interactor A", "Alt ID(s) interactor B", "Alias(es) interactor A", "Alias(es) interactor B".

In [56]:
# Columns to remove: "Taxid interactor A", "Taxid interactor B", "Alt ID(s) interactor A", "Alt ID(s) interactor B", "Alias(es) interactor A", "Alias(es) interactor B"

features_to_remove = ["Taxid interactor A", "Taxid interactor B", "Alt. ID(s) interactor A", "Alt. ID(s) interactor B", "Alias(es) interactor A", "Alias(es) interactor B", "Publication 1st author(s)"]

can_remove = True
for feature in features_to_remove:
    if feature not in data.columns:
        can_remove = False
        break

if can_remove:
    data = data.drop(columns=features_to_remove)
    
data.head()

ID(s) interactor A ID(s) interactor B  \
0   uniprotkb:Q8TAM1   uniprotkb:Q6ZW61   
1   uniprotkb:Q6ZW61   uniprotkb:Q8TAM1   
2   uniprotkb:Q8TAM1   uniprotkb:Q6ZW61   
3   uniprotkb:Q6ZW61   uniprotkb:Q8TAM1   
4   uniprotkb:Q8TAM1   uniprotkb:Q6ZW61   

                    Interaction detection method(s)  \
0  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
1  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
2  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
3  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
4               psi-mi:"MI:0071"(molecular sieving)   

                           Publication Identifier(s)  \
0  intact:EBI-15826730|doi:10.1073/pnas.091026810...   
1  intact:EBI-15826730|doi:10.1073/pnas.091026810...   
2  intact:EBI-15826730|doi:10.1073/pnas.091026810...   
3  intact:EBI-15826730|doi:10.1073/pnas.091026810...   
4  intact:EBI-15826730|doi:10.1073/pnas.091026810...   

                      Interaction type(s)  \
0  psi-mi:"MI:0915"(physical association)   
1           psi-mi:"MI:0914"(association)   
2  psi-mi:"MI:0915"(physical association)   
3           psi-mi:"MI:0914"(association)   
4  psi-mi:"MI:0915"(physical association)   

                                  Source database(s)  \
0  psi-mi:"MI:0465"(Database of Interacting Prote...   
1  psi-mi:"MI:0465"(Database of Interacting Prote...   
2  psi-mi:"MI:0465"(Database of Interacting Prote...   
3  psi-mi:"MI:0465"(Database of Interacting Prote...   
4  psi-mi:"MI:0465"(Database of Interacting Prote...   

                           Interaction identifier(s)  Confidence value(s)  
0  intact:EBI-15826767|dip:DIP-191289E|imex:IM-21...  intact-miscore:0.58  
1  intact:EBI-15826832|dip:DIP-191290E|imex:IM-21...  intact-miscore:0.58  
2  intact:EBI-15827036|dip:DIP-191289E|imex:IM-21...  intact-miscore:0.58  
3  intact:EBI-15827166|dip:DIP-191304E|imex:IM-21...  intact-miscore:0.58  
4  intact:EBI-15827219|dip:DIP-191292E|imex:IM-21...  intact-miscore:0.58

Now we replace the ids of the interactor with the name of the gene instead to facilite reading

- BBS6/MKKS: Q9NPJ1
- BBS10: Q8TAM1   
- BBS12: Q6ZW61

In [57]:
# BBS6/MKKS: Q9NPJ1, BBS10: Q8TAM1, BBS12: Q6ZW61

bbs6_id = "Q9NPJ1"
bbs10_id = "Q8TAM1"
bbs12_id = "Q6ZW61"

# Replace "ID(s) interactor A" and "ID(s) interactor B" by the gene name

# Change column name from "ID(s) interactor A" to "Gene name"
data = data.rename(columns={"ID(s) interactor A": "Gene name A", "ID(s) interactor B": "Gene name B"})

# Example start value: "uniprotkb:Q8TAM1"
data["Gene name A"] = data["Gene name A"].str.replace("uniprotkb:", "")
data["Gene name B"] = data["Gene name B"].str.replace("uniprotkb:", "")

# Remplace ids by gene names
data["Gene name A"] = data["Gene name A"].replace(bbs6_id, "BBS6")
data["Gene name B"] = data["Gene name B"].replace(bbs6_id, "BBS6")
data["Gene name A"] = data["Gene name A"].replace(bbs10_id, "BBS10")
data["Gene name B"] = data["Gene name B"].replace(bbs10_id, "BBS10")
data["Gene name A"] = data["Gene name A"].replace(bbs12_id, "BBS12")
data["Gene name B"] = data["Gene name B"].replace(bbs12_id, "BBS12")

data.head()

Gene name A Gene name B                   Interaction detection method(s)  \
0       BBS10       BBS12  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
1       BBS12       BBS10  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
2       BBS10       BBS12  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
3       BBS12       BBS10  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
4       BBS10       BBS12               psi-mi:"MI:0071"(molecular sieving)   

                           Publication Identifier(s)  \
0  intact:EBI-15826730|doi:10.1073/pnas.091026810...   
1  intact:EBI-15826730|doi:10.1073/pnas.091026810...   
2  intact:EBI-15826730|doi:10.1073/pnas.091026810...   
3  intact:EBI-15826730|doi:10.1073/pnas.091026810...   
4  intact:EBI-15826730|doi:10.1073/pnas.091026810...   

                      Interaction type(s)  \
0  psi-mi:"MI:0915"(physical association)   
1           psi-mi:"MI:0914"(association)   
2  psi-mi:"MI:0915"(physical association)   
3           psi-mi:"MI:0914"(association)   
4  psi-mi:"MI:0915"(physical association)   

                                  Source database(s)  \
0  psi-mi:"MI:0465"(Database of Interacting Prote...   
1  psi-mi:"MI:0465"(Database of Interacting Prote...   
2  psi-mi:"MI:0465"(Database of Interacting Prote...   
3  psi-mi:"MI:0465"(Database of Interacting Prote...   
4  psi-mi:"MI:0465"(Database of Interacting Prote...   

                           Interaction identifier(s)  Confidence value(s)  
0  intact:EBI-15826767|dip:DIP-191289E|imex:IM-21...  intact-miscore:0.58  
1  intact:EBI-15826832|dip:DIP-191290E|imex:IM-21...  intact-miscore:0.58  
2  intact:EBI-15827036|dip:DIP-191289E|imex:IM-21...  intact-miscore:0.58  
3  intact:EBI-15827166|dip:DIP-191304E|imex:IM-21...  intact-miscore:0.58  
4  intact:EBI-15827219|dip:DIP-191292E|imex:IM-21...  intact-miscore:0.58

### <b>Filter data tuples</b>

Remove tuples with a miscore < 0.4

In [58]:
# Remove tuples with values < 0.4

"""
Example values of "Confidence value(s)" column:
- "intact-miscore:0.56|author-based confidence:0.868278297126"
- "intact-miscore:0.56"

Not all rows have the "author-based confidence" value
"""

print(f"Number of rows before filtering >0.4 miscore: {len(data)}")

# 1. Create column "IntAct Miscore" 
data["IntAct Miscore"] = data["Confidence value(s)"].str.extract(r"intact-miscore:(\d\.\d+)")
data["IntAct Miscore"] = data["IntAct Miscore"].astype(float)

# 2. Create column "Autor-based confidence"
data["Autor-based confidence"] = data["Confidence value(s)"].str.extract(r"author-based confidence:(\d\.\d+)")
data["Autor-based confidence"] = data["Autor-based confidence"].astype(float)

# 3. Remove "Confidence value(s)" column
data = data.drop(columns=["Confidence value(s)"])

# 4. Remove tuples with values < 0.4
data = data[data["IntAct Miscore"] >= 0.4]

print(f"Number of rows after filtering >0.4 miscore: {len(data)}")

Number of rows before filtering >0.4 miscore: 105
Number of rows after filtering >0.4 miscore: 82


In [59]:
data.head()

Gene name A Gene name B                   Interaction detection method(s)  \
0       BBS10       BBS12  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
1       BBS12       BBS10  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
2       BBS10       BBS12  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
3       BBS12       BBS10  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
4       BBS10       BBS12               psi-mi:"MI:0071"(molecular sieving)   

                           Publication Identifier(s)  \
0  intact:EBI-15826730|doi:10.1073/pnas.091026810...   
1  intact:EBI-15826730|doi:10.1073/pnas.091026810...   
2  intact:EBI-15826730|doi:10.1073/pnas.091026810...   
3  intact:EBI-15826730|doi:10.1073/pnas.091026810...   
4  intact:EBI-15826730|doi:10.1073/pnas.091026810...   

                      Interaction type(s)  \
0  psi-mi:"MI:0915"(physical association)   
1           psi-mi:"MI:0914"(association)   
2  psi-mi:"MI:0915"(physical association)   
3           psi-mi:"MI:0914"(association)   
4  psi-mi:"MI:0915"(physical association)   

                                  Source database(s)  \
0  psi-mi:"MI:0465"(Database of Interacting Prote...   
1  psi-mi:"MI:0465"(Database of Interacting Prote...   
2  psi-mi:"MI:0465"(Database of Interacting Prote...   
3  psi-mi:"MI:0465"(Database of Interacting Prote...   
4  psi-mi:"MI:0465"(Database of Interacting Prote...   

                           Interaction identifier(s)  IntAct Miscore  \
0  intact:EBI-15826767|dip:DIP-191289E|imex:IM-21...            0.58   
1  intact:EBI-15826832|dip:DIP-191290E|imex:IM-21...            0.58   
2  intact:EBI-15827036|dip:DIP-191289E|imex:IM-21...            0.58   
3  intact:EBI-15827166|dip:DIP-191304E|imex:IM-21...            0.58   
4  intact:EBI-15827219|dip:DIP-191292E|imex:IM-21...            0.58   

   Autor-based confidence  
0                     NaN  
1                     NaN  
2                     NaN  
3                     NaN  
4                     NaN

### <b>Save data</b>

Into csv file

In [60]:
# Save data

cleandata_filename = "intact_chaperinin_data.csv"

data.to_csv(data_source + cleandata_filename, index=False)

### <b>Subsequent analysis</b>

- Visualize the gene combinations that have the most number of interactions
- Sort interactions by higher MIscore

In [61]:
# Visualize the gene combinations that have the most number of interactions

# Count unique interactions for each gene
interaction_counts = data.groupby(['Gene name A', 'Gene name B']).size().reset_index(name='Interaction Count')

# Display top interacting genes for initial insights
top_interactions = interaction_counts.sort_values(by='Interaction Count', ascending=False).head(10)
top_interactions

Gene name A Gene name B  Interaction Count
11       BBS12        BBS6                 11
13       BBS12      Q8IWZ6                  7
24        BBS6      Q8IWZ6                  5
19        BBS6       BBS12                  4
30      Q8IWZ6       BBS10                  3
22        BBS6      P78371                  3
21        BBS6      P17987                  3
1        BBS10       BBS12                  3
26      O60232       BBS10                  3
28      P26641        BBS6                  2

In [ ]:
# Sort the interactions by higher miscore
    
top_interactions = data.sort_values(by='IntAct Miscore', ascending=False).head(10)
top_interactions

Gene name A Gene name B                   Interaction detection method(s)  \
18        BBS6       BBS12  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
10       BBS12        BBS6  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
21       BBS12        BBS6  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
20       BBS12        BBS6  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
19       BBS12        BBS6  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
16       BBS12        BBS6  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
15       BBS12        BBS6  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
13        BBS6       BBS12  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
12       BBS12        BBS6  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
14        BBS6       BBS12  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   

                            Publication Identifier(s)  \
18   intact:EBI-6127424|pubmed:22500027|imex:IM-17327   
10  intact:EBI-15826730|doi:10.1073/pnas.091026810...   
21   intact:EBI-6127424|pubmed:22500027|imex:IM-17327   
20   intact:EBI-6127424|pubmed:22500027|imex:IM-17327   
19   intact:EBI-6127424|pubmed:22500027|imex:IM-17327   
16  intact:EBI-27103963|doi:10.1016/j.cell.2021.04...   
15  intact:EBI-15481134|doi:10.1038/nature22366|pu...   
13  intact:EBI-21404059|pubmed:26900326|imex:IM-27022   
12  intact:EBI-15826730|doi:10.1073/pnas.091026810...   
14  intact:EBI-21404059|pubmed:26900326|imex:IM-27022   

                       Interaction type(s)  \
18           psi-mi:"MI:0914"(association)   
10  psi-mi:"MI:0915"(physical association)   
21           psi-mi:"MI:0914"(association)   
20           psi-mi:"MI:0914"(association)   
19           psi-mi:"MI:0914"(association)   
16           psi-mi:"MI:0914"(association)   
15           psi-mi:"MI:0914"(association)   
13  psi-mi:"MI:0915"(physical association)   
12  psi-mi:"MI:0915"(physical association)   
14  psi-mi:"MI:0915"(physical association)   

                                   Source database(s)  \
18  psi-mi:"MI:0469"(European Bioinformatics Insti...   
10  psi-mi:"MI:0465"(Database of Interacting Prote...   
21  psi-mi:"MI:0469"(European Bioinformatics Insti...   
20  psi-mi:"MI:0469"(European Bioinformatics Insti...   
19  psi-mi:"MI:0469"(European Bioinformatics Insti...   
16  psi-mi:"MI:0469"(European Bioinformatics Insti...   
15  psi-mi:"MI:0469"(European Bioinformatics Insti...   
13  psi-mi:"MI:0469"(European Bioinformatics Insti...   
12  psi-mi:"MI:0465"(Database of Interacting Prote...   
14  psi-mi:"MI:0469"(European Bioinformatics Insti...   

                            Interaction identifier(s)  IntAct Miscore  \
18                intact:EBI-6128347|imex:IM-17327-34            0.83   
10  intact:EBI-15827189|dip:DIP-191297E|imex:IM-21...            0.83   
21                intact:EBI-6128507|imex:IM-17327-39            0.83   
20                intact:EBI-6128421|imex:IM-17327-37            0.83   
19                intact:EBI-6128394|imex:IM-17327-36            0.83   
16                                intact:EBI-54795348            0.83   
15             intact:EBI-21894918|imex:IM-25778-6979            0.83   
13                intact:EBI-21404081|imex:IM-27022-1            0.83   
12  intact:EBI-15827266|dip:DIP-191297E|imex:IM-21...            0.83   
14                intact:EBI-21404108|imex:IM-27022-3            0.83   

    Autor-based confidence  
18                     NaN  
10                     NaN  
21                     NaN  
20                     NaN  
19                     NaN  
16                     NaN  
15                     NaN  
13                     NaN  
12                     NaN  
14                     NaN

In [67]:
# Sort the interactions by higher miscore again but this time ignore the interactions between (bbs6, bbs12) and (bbs12, bbs6)

data_without_bbs6_bbs12 = data[(data["Gene name A"] != "BBS6") | (data["Gene name B"] != "BBS12")]
data_without_bbs6_bbs12 = data_without_bbs6_bbs12[(data_without_bbs6_bbs12["Gene name A"] != "BBS12") | (data_without_bbs6_bbs12["Gene name B"] != "BBS6")]

top_interactions = data_without_bbs6_bbs12.sort_values(by='IntAct Miscore', ascending=False).head(10)
top_interactions

Gene name A Gene name B                   Interaction detection method(s)  \
54       BBS12      Q8IWZ6  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
49       BBS12      Q8IWZ6  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
58       BBS12      Q8IWZ6  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
57      Q8IWZ6       BBS12  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
56       BBS12      Q8IWZ6  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
55       BBS12      Q8IWZ6  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
35       BBS12      Q8IWZ6  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
50       BBS12      Q8IWZ6  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
61      Q8IWZ6       BBS10  psi-mi:"MI:0007"(anti tag coimmunoprecipitation)   
37       BBS10      Q8IWZ6               psi-mi:"MI:0071"(molecular sieving)   

                            Publication Identifier(s)  \
54   intact:EBI-6127424|pubmed:22500027|imex:IM-17327   
49  intact:EBI-15481134|doi:10.1038/nature22366|pu...   
58   intact:EBI-6127424|pubmed:22500027|imex:IM-17327   
57   intact:EBI-6127424|pubmed:22500027|imex:IM-17327   
56   intact:EBI-6127424|pubmed:22500027|imex:IM-17327   
55   intact:EBI-6127424|pubmed:22500027|imex:IM-17327   
35  intact:EBI-15826730|doi:10.1073/pnas.091026810...   
50  intact:EBI-27103963|doi:10.1016/j.cell.2021.04...   
61  intact:EBI-27103963|doi:10.1016/j.cell.2021.04...   
37  intact:EBI-15826730|doi:10.1073/pnas.091026810...   

                       Interaction type(s)  \
54  psi-mi:"MI:0915"(physical association)   
49           psi-mi:"MI:0914"(association)   
58           psi-mi:"MI:0914"(association)   
57  psi-mi:"MI:0915"(physical association)   
56           psi-mi:"MI:0914"(association)   
55           psi-mi:"MI:0914"(association)   
35  psi-mi:"MI:0915"(physical association)   
50           psi-mi:"MI:0914"(association)   
61           psi-mi:"MI:0914"(association)   
37  psi-mi:"MI:0915"(physical association)   

                                   Source database(s)  \
54  psi-mi:"MI:0469"(European Bioinformatics Insti...   
49  psi-mi:"MI:0469"(European Bioinformatics Insti...   
58  psi-mi:"MI:0469"(European Bioinformatics Insti...   
57  psi-mi:"MI:0469"(European Bioinformatics Insti...   
56  psi-mi:"MI:0469"(European Bioinformatics Insti...   
55  psi-mi:"MI:0469"(European Bioinformatics Insti...   
35  psi-mi:"MI:0465"(Database of Interacting Prote...   
50  psi-mi:"MI:0469"(European Bioinformatics Insti...   
61  psi-mi:"MI:0469"(European Bioinformatics Insti...   
37  psi-mi:"MI:0465"(Database of Interacting Prote...   

                            Interaction identifier(s)  IntAct Miscore  \
54                intact:EBI-6128382|imex:IM-17327-35            0.78   
49             intact:EBI-21894918|imex:IM-25778-6979            0.78   
58                intact:EBI-6128507|imex:IM-17327-39            0.78   
57                intact:EBI-6128486|imex:IM-17327-38            0.78   
56                intact:EBI-6128421|imex:IM-17327-37            0.78   
55                intact:EBI-6128394|imex:IM-17327-36            0.78   
35  intact:EBI-15827121|dip:DIP-191311E|imex:IM-21...            0.78   
50                                intact:EBI-54795348            0.78   
61                                intact:EBI-54502780            0.75   
37  intact:EBI-15827219|dip:DIP-191292E|imex:IM-21...            0.75   

    Autor-based confidence  
54                     NaN  
49                     NaN  
58                     NaN  
57                     NaN  
56                     NaN  
55                     NaN  
35                     NaN  
50                     NaN  
61                     NaN  
37                     NaN

In [68]:
top_interactions[['Gene name A', 'Gene name B', 'IntAct Miscore']]

Gene name A Gene name B  IntAct Miscore
54       BBS12      Q8IWZ6            0.78
49       BBS12      Q8IWZ6            0.78
58       BBS12      Q8IWZ6            0.78
57      Q8IWZ6       BBS12            0.78
56       BBS12      Q8IWZ6            0.78
55       BBS12      Q8IWZ6            0.78
35       BBS12      Q8IWZ6            0.78
50       BBS12      Q8IWZ6            0.78
61      Q8IWZ6       BBS10            0.75
37       BBS10      Q8IWZ6            0.75